mainwindow.cpp

```diff
@@ -488,21 +497,77 @@ bool MainWindow::eventFilter(QObject *watched, QEvent *event)
     return false;
 }
 
-void MainWindow::on_actionStart_Attack_triggered()
+void MainWindow::startAttack()
 {
     if (!checkSettings())
         return;
 
+    QStringList parameters = getAttackParameters();
+
+    // Session for johnny
+    QString nameOfFile = m_session + ".rec";
+    
+    if (QFileInfo(nameOfFile).isReadable()) {
+        int button = QMessageBox::question(
+            this,
+            tr("Johnny"),
+            tr("Johnny is about to overwrite your previous session file. Do you want to proceed?"),
+            QMessageBox::Yes | QMessageBox::No);
+       if (button == QMessageBox::No)
+            return;
+        // Remove .rec file to avoid problem when john does not write it.
+        if(!QFile(nameOfFile).remove())
+        {
+            QMessageBox::warning(
+                this,
+                tr("Warning"),
+                tr("Unable to remove file ") + nameOfFile);
+        }
+    }
+
+    QFile description(m_session + ".johnny");
+    if (!description.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        QMessageBox::critical(
+            this,
+            tr("Johnny"),
+            tr("Johnny could not open file to save session description!"));
+        return;
+    }
+    QTextStream descriptionStream(&description);
+
+    for(int i = 0; i < m_hashesFilesNames.size(); i++)
+    {
+        descriptionStream << "FILE=" << m_hashesFilesNames[i] <<endl;
+    }
+    descriptionStream << "FORMAT=" << m_format << endl;
+    description.close();
+
+    parameters << QString("--session=%1").arg(m_session);
+
+    // We check that we have file name.
+    if (!m_hashesFilesNames.isEmpty()) {
+        // If file name is not empty then we have file, pass it to
+        // John.
+        // We add file name onto parameters list.
+        parameters << m_hashesFilesNames;
+        startJohn(parameters);
+    } else {
+        QMessageBox::critical(
+            this,
+            tr("Johnny"),
+            tr("Johnny don't have access to this file. Did you forget to save it ?"));
```

Comment:
I know that your primary language is French, so I suggest you correct your spelling in a word processor before adding language text to code. @math07